linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -645,46 +627,14 @@ private[emitter] object CoreJSLib {
           } else {
             genBodyNoSwitch(maybeWithoutLong)
           }
-        }
-
-        defineDispatcher(methodName, args, {
-          genBodyMaybeSwitch()
         })
       }
 
-      val rhs = varRef("rhs")
-
-      defineStandardDispatcher(equalsMethodName, varRef("rhs") :: Nil,
-          Some(instance === rhs),
-          List(BoxedDoubleClass, BoxedLongClass, BoxedCharacterClass))
-
-      defineStandardDispatcher(hashCodeMethodName, Nil,
-          Some(genCallHelper("systemIdentityHashCode", instance)),
-          List(BoxedStringClass, BoxedDoubleClass, BoxedBooleanClass,
-              BoxedUnitClass, BoxedLongClass, BoxedCharacterClass))
-
-      defineStandardDispatcher(compareToMethodName, varRef("rhs") :: Nil,
-          None,
-          List(BoxedStringClass, BoxedDoubleClass, BoxedBooleanClass,
-              BoxedLongClass, BoxedCharacterClass))
-
-      defineStandardDispatcher(lengthMethodName, Nil, None,
-          List(BoxedStringClass))
-
-      defineStandardDispatcher(charAtMethodName, varRef("index") :: Nil, None,
-          List(BoxedStringClass))
-
-      defineStandardDispatcher(subSequenceMethodName,
-          varRef("start") :: varRef("end") :: Nil, None,
-          List(BoxedStringClass))
-
       for {
-        methodName <- List(byteValueMethodName, shortValueMethodName,
-            intValueMethodName, longValueMethodName, floatValueMethodName,
-            doubleValueMethodName)
+        (methodName, representativeClasses) <- globalKnowledge.getRepresentativeClassMethods()
```

Comment:
That's neat, but it's a bit *too* neat. There are a number of things that should be pruned from that list:

* Methods that are only defined in hijacked classes themselves, but none of their super types. In particular, most methods of `String`. Those will never be called through a dispatcher, because the static type is needed to call them, and hence `FunctionEmitter` can generate a direct call to the correct implementation. (This is an optimization.)
* `Byte`, `Short`, `Integer` and `Float` must be filtered out, because `Double` takes care of them. This is required for correctness, because for example `Integer.compareTo(Object)` casts its argument to `int`, which fails if the left-hand-side is `5.0` and the rhs is `5.5`, for example.